src/libcore/ptr/unique.rs

```diff
@@ -184,6 +190,7 @@ impl<T: ?Sized> From<&T> for Unique<T> {
 impl<T: ?Sized> From<NonNull<T>> for Unique<T> {
     #[inline]
     fn from(p: NonNull<T>) -> Self {
+        // SAFETY: A NonNull pointer cannot be null
         unsafe { Unique::new_unchecked(p.as_ptr()) }
```

Comment:
Similarly to the &T note by @ecstatic-morse -- I think this is not okay, Unique (at least in principle) has stricter requirements than NonNull.

Reply:
Agreed, this instance seems wrong.

Reply:
Looks like it was added [here](https://github.com/rust-lang/rust/commit/fb03a49c2501c52401b3c987fd455818de1736f2#diff-9839f9730ca3a90d5c0ebb0a7edd67cf) as part of https://github.com/rust-lang/rust/pull/46952.

Reply:
I removed the comment for now